pkgs/build-support/setup-hooks/verify-signature.sh

```diff
@@ -0,0 +1,45 @@
+# Helper functions for verifying (detached) PGP signatures
+
+# importPublicKey
+# Add PGP public key contained in ${publicKey} to the keyring.
+# All imported keys will be trusted by verifySig
+_importPublicKey() {
+  if [ -z "$signaturePublicKey" ]; then
+    echo "error: verifySignatureHook requires signaturePublicKey" >&2
+    exit 1
+  fi
+  gpg -q --import "$signaturePublicKey"
+}
+
+
+# verifySignature SIGFILE DATAFILE UNCOMPRESS
+# verify the signature SIGFILE for the file DATAFILE
+# if DATAFILE is omitted, it is derived from SIGFILE by dropping the .asc or .sig suffix
+# if UNCOMPRESS is set, uncompress DATAFILE before verification
+verifySignature() {
+  if [ -z "$3" ]; then
+    gpgv --keyring pubring.kbx "$1" "$2" || exit 1
```

Comment:
Can you make gpg ignoring expire dates? Otherwise it will become a problem when somebody tries to build an old nixpkgs revision. We could print a warning if a gpg key is expired.

Reply:
The warning should be good enough for the TOFU case.